backend/manager/modules/common/src/main/java/org/ovirt/engine/core/common/businessentities/comparators/VmsCpuPinningPolicyComparator.java

```diff
@@ -0,0 +1,28 @@
+package org.ovirt.engine.core.common.businessentities.comparators;
+
+import java.io.Serializable;
+import java.util.Comparator;
+
+import org.ovirt.engine.core.common.businessentities.VM;
+
+/**
+ * Comparing VMs based on the CPU pinning policy and if needed, their CPU topology.
+ */
+public class VmsCpuPinningPolicyComparator implements Comparator<VM>, Serializable {
+
+    @Override
+    public int compare(VM o1, VM o2) {
+        // Check the policy, their values are from the higher (more restricted policy) to the lower.
+        if (o2.getCpuPinningPolicy().getValue() - o1.getCpuPinningPolicy().getValue() != 0) {
+            return o2.getCpuPinningPolicy().getValue() - o1.getCpuPinningPolicy().getValue();
+        }
+        // The policies are equal, compare based on CPU topology.
```

Comment:
Shouldn't we start with pinning of those VMs that require the max threads per core? That seems like the biggest restriction on the topology.

Reply:
Let's combine the below conversion with this as well. As we mainly discuss on how to compare.